src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    public MachineServiceImpl() {
+    }
+
+    public List<Machine> getAll(Class type) {
+        List<Machine> result = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            result = (List<Machine>) new BulldozerProducer().get();
```

Comment:
```suggestion
            return new ArrayList<>(bulldozerProducer.get());
```
спробуй так спростити, компілятор мав би пропустити. 
Доречі, коли ти в методі викликаєш new BulldozerProducer() - створюєш новий об'єкт продюсера. Загалом зараз нам це робити не потрібно адже нам достатньо одного об'єкта на клас і нам ньому можна викликати метод get(). Давай винесемо продюсера як змінну класу